[Messenger] Symfony 7.4.0 FailedMessagesRetryCommand does not work with SigningSerializer

### Symfony version(s) affected

7.4.0

### Description

I've tried upgrading our app from 7.3 to 7.4.

`bin/console lint:container` fails with the following error in our CI:

```
 [ERROR] Invalid definition for service "console.command.messenger_failed_messages_retry": argument 6 of                
         "Symfony\Component\Messenger\Command\FailedMessagesRetryCommand::__construct()" accepts                        
         "Symfony\Component\Messenger\Transport\Serialization\PhpSerializer",                                           
         "Symfony\Component\Messenger\Transport\Serialization\SigningSerializer" passed. 
```

Running the `messenger:failed:retry` command directly produces the same result:

```
TypeError {#75222
  #message: "Symfony\Component\Messenger\Command\FailedMessagesRetryCommand::__construct(): Argument #6 ($phpSerializer) must be of type ?Symfony\Component\Messenger\Transport\Serialization\PhpSerializer, Symfony\Component\Messenger\Transport\Serialization\SigningSerializer given, called in /srv/app/var/cache/dev/ContainerXyCsCQA/getConsole_Command_MessengerFailedMessagesRetryService.php on line 26"                                
  #code: 0
  #file: "./vendor/symfony/messenger/Command/FailedMessagesRetryCommand.php"
  #line: 49
  -locals: ? array
```

In the built container the `console.command.messenger_failed_messages_retry` service now receives the `.signing.messenger.transport.native_php_serializer` service which is a `SigningSerializer` instance that decorates the `PhpSerializer` instance, but the type in the `FailedMessagesRetryCommand` command constructor allows only `PhpSerializer` instances.

```
# this is how .signing.messenger.transport.native_php_serializer looks in the container which is what is being passed as an argument to the FailedMessagesRetryCommand command

        return $container->privates['.signing.messenger.transport.native_php_serializer'] = new \Symfony\Component\Messenger\Transport\Serialization\SigningSerializer(new \Symfony\Component\Messenger\Transport\Serialization\PhpSerializer(), \Symfony\Component\String\LazyString::fromCallable(#[\Closure(name: '.messenger.signing_serializer.signing_key', class: 'string')] fn () => ($container->privates['.messenger.signing_serializer.signing_key'] ?? $container->load('get_Messenger_SigningSerializer_SigningKeyService'))), []);
```


### How to reproduce

Just run the `messenger:failed:retry` command in any 7.4 app which uses Messenger for processing messages.

### Possible Solution

_No response_

### Additional Context

_No response_

--- a/src/Symfony/Bundle/FrameworkBundle/Resources/config/console.php
+++ b/src/Symfony/Bundle/FrameworkBundle/Resources/config/console.php
@@ -201,7 +201,7 @@ return static function (ContainerConfigurator $container) {
                 service('messenger.routable_message_bus'),
                 service('event_dispatcher'),
                 service('logger')->nullOnInvalid(),
-                service('messenger.transport.native_php_serializer')->nullOnInvalid(),
+                service('.messenger.transport.native_php_serializer')->nullOnInvalid(),
                 null,
             ])
             ->tag('console.command')
@@ -211,7 +211,7 @@ return static function (ContainerConfigurator $container) {
             ->args([
                 abstract_arg('Default failure receiver name'),
                 abstract_arg('Receivers'),
-                service('messenger.transport.native_php_serializer')->nullOnInvalid(),
+                service('.messenger.transport.native_php_serializer')->nullOnInvalid(),
             ])
             ->tag('console.command')
 
@@ -219,7 +219,7 @@ return static function (ContainerConfigurator $container) {
             ->args([
                 abstract_arg('Default failure receiver name'),
                 abstract_arg('Receivers'),
-                service('messenger.transport.native_php_serializer')->nullOnInvalid(),
+                service('.messenger.transport.native_php_serializer')->nullOnInvalid(),
             ])
             ->tag('console.command')
 

--- a/src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.php
+++ b/src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.php
@@ -78,7 +78,10 @@ return static function (ContainerConfigurator $container) {
         ->set('serializer.normalizer.flatten_exception', FlattenExceptionNormalizer::class)
             ->tag('serializer.normalizer', ['built_in' => true, 'priority' => -880])
 
+        ->set('.messenger.transport.native_php_serializer', PhpSerializer::class)
         ->set('messenger.transport.native_php_serializer', PhpSerializer::class)
+            ->factory('current')
+            ->args([[service('.messenger.transport.native_php_serializer')]])
         ->alias('messenger.default_serializer', 'messenger.transport.native_php_serializer')
         ->alias(SerializerInterface::class, 'messenger.default_serializer')
 

